airbyte-integrations/connectors/source-recharge/source_recharge/api.py

```diff
@@ -56,6 +56,15 @@ def get_stream_data(self, response_data: Any) -> List[dict]:
         else:
             return [response_data]
 
+    def should_retry(self, response: requests.Response) -> bool:
+        res = super().should_retry(response)
+        if res:
+            return res
+
+        # For some reason for large responses API server can return non-complete result
+        content_length = int(response.headers.get('Content-Length', 0))
+        return bool(content_length) and content_length > len(response.content)
```

Comment:
Can we check a code value of responses? Maybe the resource returns 204 code for this case? And is there some information or docs about empty responses?

Reply:
No, this not related to 204 at all.
Response status = 200 but data is incomplete
Tt looks like some HTTP server layer problem
I have re-implemented and added check for status = 200 to be more explicit for this specific case

Reply:
Confirm @grubberr the response status is 200